spaceship/lib/spaceship/connect_api/models/app_clip_header_image.rb

```diff
@@ -0,0 +1,173 @@
+require_relative '../file_uploader'
+require_relative '../model'
+require 'spaceship/globals'
+
+module Spaceship
+  class ConnectAPI
+    class AppClipHeaderImage
```

Comment:
This class's implementation is almost identical to `AppScreenshot` and `AppPreview` but I didn't go refactoring to extract common interfaces for now. 

I personally feel like -

* Having the list of `attr_accesor` in each class should be fine
* `awaiting_upload?` or other helpers functions can be extracted to a module to be shared
* Asset uploading logic should be extracted to a class to enforce exactly the same flow for each but that needs a bit of abstraction since APIs used are different